### ISSUE TITLE
[RLlib] SAC potentially has a memory leak

### ISSUE THREAD

**Issue by subasah** [ray-project/ray#19770]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

When I run the SAC algorithm for my custom environment. The PPO runs fine with the current configuration of the training but SAC is terminating. I can see high RAM utilization by the SAC hence probably terminating. 

I don't think this is the expected behavior of the SAC algorithm. 
Any recommendation will be appreciated. 
 

### Versions / Dependencies

Ray version: 1.4.0
Python Version: 3.7.10
Tensorflow Version: 1.14.0
OS: Fedora 34

### Reproduction script

    from NR_IES.envs.NR_IES_env import NR_IES_v0
    from ray.tune.registry import register_env
    import gym
    import os
    import ray
    from ray.rllib.agents.sac.sac import SACTrainer, DEFAULT_CONFIG
    import shutil
    from ray import tune
    import logging

     def main ():

      chkpt_root = "sac_training_26oct/NR_IES"
      shutil.rmtree(chkpt_root, ignore_errors=True, onerror=None)
      ray.init(local_mode=True, include_dashboard= False, logging_level=logging.DEBUG)
  
      # custom environment registration
      select_env = "NR_IES-v0"
      register_env(select_env, lambda config: NR_IES_v0())
  
      # create agent and environment configuration
      config = DEFAULT_CONFIG.copy()
      config["log_level"] = "WARN"
      config['num_workers'] = 18
      config['num_gpus'] = 1
      config['num_gpus_per_worker'] = (1-0.0001)/18
  
      agent = SACTrainer(config, env=select_env)
      
      status = "{:2d} reward {:6.2f}/{:6.2f}/{:6.2f} len {:4.2f} saved {}"
      n_iter = 1000000
  
      for n in range(n_iter):
          result = agent.train()
          chkpt_file = agent.save(chkpt_root)
          print(status.format(
                  n + 1,
                  result["episode_reward_min"],
                  result["episode_reward_mean"],
                  result["episode_reward_max"],
                  result["episode_len_mean"],
                  chkpt_file
                  ))
  
      if __name__ == "__main__":
          main()

### Anything else

Configuration I used to scale my training is:
https://docs.ray.io/en/latest/rllib-training.html#scaling-guide

Log for SAC failure can be seen here. 
https://pastebin.com/Q2SuG1yA

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by avnishn** [comment]
I'm going to close this due to inactivity. Feel free to reopen if you have any additional questions/comments
